scalekit/core.py

```
@@ -254,15 +328,24 @@ def grpc_exec(
                     raise ScalekitServerException.promote(exp)
                 try:
                     self.__authenticate_client()
-                    return self.grpc_exec(func, data, retry=retry-1)
+                    return self.grpc_exec(func, data, retry=retry - 1, attempt=attempt + 1, timeout=timeout)
                 except Exception as refresh_exp:
                     raise ScalekitServerException.promote(exp)
             elif exp.code() == grpc.StatusCode.RESOURCE_EXHAUSTED:
                 # Surface Scalekit rate-limits immediately — retrying triples the damage
                 raise ScalekitServerException.promote(exp)
```

Ruff (0.16.3)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)